--- source/main/utils/InputEngine.cpp
+++ source/main/utils/InputEngine.cpp
@@ -395,9 +395,5 @@
     {"", -1, "", ""},
 };
 
-#if OGRE_PLATFORM == OGRE_PLATFORM_APPLE
-#define strnlen(str,len) strlen(str)
-#endif
-
 //Use this define to signify OIS will be used as a DLL
 //(so that dll import/export macros are in effect)
@@ -492,6 +488,8 @@
         pl.insert(OIS::ParamList::value_type("XAutoRepeatOn", "false"));
         pl.insert(OIS::ParamList::value_type("x11_mouse_grab", "false"));
         pl.insert(OIS::ParamList::value_type("x11_keyboard_grab", "false"));
+#elif OGRE_PLATFORM == OGRE_PLATFORM_APPLE
+        pl.insert(OIS::ParamList::value_type("MacAutoRepeatOn", "false"));
 #else
         pl.insert(OIS::ParamList::value_type("w32_mouse", "DISCL_FOREGROUND"));
         pl.insert(OIS::ParamList::value_type("w32_mouse", "DISCL_NONEXCLUSIVE"));
