--- source/main/AppContext.cpp
+++ source/main/AppContext.cpp
@@ -55,6 +55,10 @@
 #include <string>
 #include <ctime>
 
+#if OGRE_PLATFORM == OGRE_PLATFORM_APPLE
+#include "MacOS.h"
+#endif
+
 using namespace RoR;
 
 // --------------------------
@@ -563,6 +567,13 @@
         process_dir = "/usr/share/rigsofrods/resources/";
     }
 #endif
+#if OGRE_PLATFORM == OGRE_PLATFORM_APPLE
+    std::string bundle_res = RoR::MacOS::GetBundleResourcesDir();
+    if (!bundle_res.empty())
+    {
+        process_dir = bundle_res;
+    }
+#endif
     if (!FolderExists(process_dir))
     {
         ErrorUtils::ShowError(_L("Startup error"), _L("Resources folder not found. Check if correctly installed."));
